### PR TITLE
adding alternative of get_memebership API

### PR DESCRIPTION
**Pull request by abimishr**
1. **Git PR Link:** https://github.com/madhansansel/dnacenter-ansible/pull/294
2.  **Problem Description:** There is a potential error when we try to fetch the list using get_membership API to see whether a device is associated with a site.
3.  **Root Cause:** The Get_site_assignment function should handle cases where the relevant data might be missing or None. It should either gracefully return a default value or raise a more informative exception to indicate the issue.
4.  **Code Changes:** Have added a new method which uses the API 'get_device_detail'

### CODE REVIEW

File: plugins/modules/provision_workflow_manager.py [hunk 1]
```diff
@@ -615,7 +615,41 @@ def get_site_details(self, site_name_hierarchy=None):
...
+        Tells us the whether a device is assigned to the site
+
+        Parameters:
+          - self: The instance of the class containing the 'config' attribute
+                  to be validated.
+        Returns:
+          - boolean: True if any device is associated with the site, False if no device is associated with site
+
+        Example:
+          Post creation of the validated input, this method tells whether devices are associated with a site.
```
Comment (madhansansel) [inline plugins/modules/provision_workflow_manager.py:629]: This example is not required.

File: plugins/modules/provision_workflow_manager.py [hunk 2]
```diff
@@ -615,7 +615,41 @@ def get_site_details(self, site_name_hierarchy=None):
...
+        """
+
+        self.log("Device ID of the device is {0}".format(uuid), "INFO")
```
Comment (madhansansel) [inline plugins/modules/provision_workflow_manager.py:632]: Checking site assignment for device with UUID: {0}

File: plugins/modules/provision_workflow_manager.py [hunk 3]
```diff
@@ -615,7 +615,41 @@ def get_site_details(self, site_name_hierarchy=None):
...
+                op_modifies=True
+            )
+            self.log("Response collected from the API 'get_device_detail' {0}".format(site_response))
+            site_response = site_response.get("response")
+            if site_response.get("location"):
+                return True
+            else:
+                return False
+        except Exception as e:
+            msg = "Device with uuid {0} not found due to {1}".format(uuid, e)
```
Comment (madhansansel) [inline plugins/modules/provision_workflow_manager.py:648]: msg = "Failed to find device with UUID {0} due to: {1}"

File: plugins/modules/provision_workflow_manager.py [hunk 4]
```diff
@@ -1056,8 +1091,9 @@ def verify_diff_merged(self):
         device_type = self.want.get("device_type")
         provisioning = self.validated_config[0].get("provisioning")
         site_name_hierarchy = self.validated_config[0].get("site_name_hierarchy")
+        uuid = self.get_device_id()
         if provisioning is False:
-            if self.get_site_assignment() is True:
+            if self.get_site_assignment(uuid=uuid) is True:
```
Comment (madhansansel) [inline plugins/modules/provision_workflow_manager.py:1096]: we can directly write as 
`if self.get_site_assignment(uuid)`

File: plugins/modules/provision_workflow_manager.py [hunk 5]
```diff
@@ -615,7 +615,41 @@ def get_site_details(self, site_name_hierarchy=None):
 
         return (site_exists, site_id)
 
-    def get_site_assignment(self):
+    def get_site_assignment(self, uuid=None):
```
[the remark this answers is not in the document]
Comment (abimishr) [reply plugins/modules/provision_workflow_manager.py]: Yeah, I have made it mandatory

File: plugins/modules/provision_workflow_manager.py [hunk 6]
```diff
@@ -615,7 +615,41 @@ def get_site_details(self, site_name_hierarchy=None):
...
+        """
+        Tells us the whether a device is assigned to the site
```
[the remark this answers is not in the document]
Comment (abimishr) [reply plugins/modules/provision_workflow_manager.py]: Changed

File: plugins/modules/provision_workflow_manager.py [hunk 7]
```diff
@@ -615,7 +615,41 @@ def get_site_details(self, site_name_hierarchy=None):
...
+
+        Parameters:
+          - self: The instance of the class containing the 'config' attribute
+                  to be validated.
```
[the remark this answers is not in the document]
Comment (abimishr) [reply plugins/modules/provision_workflow_manager.py:624]: Optional part is removed

File: plugins/modules/provision_workflow_manager.py [hunk 8]
```diff
@@ -615,7 +615,41 @@ def get_site_details(self, site_name_hierarchy=None):
...
+        Returns:
+          - boolean: True if any device is associated with the site, False if no device is associated with site
```
[the remark this answers is not in the document]
Comment (abimishr) [reply plugins/modules/provision_workflow_manager.py]: Addressed

File: plugins/modules/provision_workflow_manager.py (hunk 1 above)
Comment (abimishr) [reply plugins/modules/provision_workflow_manager.py]: Removed

File: plugins/modules/provision_workflow_manager.py (hunk 2 above)
Comment (abimishr) [reply plugins/modules/provision_workflow_manager.py]: Changed

File: plugins/modules/provision_workflow_manager.py (hunk 3 above)
Comment (abimishr) [reply plugins/modules/provision_workflow_manager.py]: Addressed

File: plugins/modules/provision_workflow_manager.py (hunk 4 above)
Comment (abimishr) [reply plugins/modules/provision_workflow_manager.py]: Done

**Review by madhansansel — APPROVED** [review]

**Pull request merged**